plugins/net_plugin/net_plugin.cpp

```diff
@@ -3942,16 +3962,31 @@ namespace eosio {
                                std::chrono::seconds( options.at("connection-cleanup-period").as<int>() ),
                                options.at("max-clients").as<uint32_t>() );
 
-         if( options.count( "p2p-listen-endpoint" ) && options.at("p2p-listen-endpoint").as<string>().length()) {
-            p2p_address = options.at( "p2p-listen-endpoint" ).as<string>();
-            EOS_ASSERT( p2p_address.length() <= max_p2p_address_length, chain::plugin_config_exception,
-                        "p2p-listen-endpoint too long, must be less than ${m}", ("m", max_p2p_address_length) );
+         if( options.count( "p2p-listen-endpoint" ) && !options.at("p2p-listen-endpoint").as<vector<string>>().empty() &&  options.at("p2p-listen-endpoint").as<vector<string>>()[0].length()) {
```

Comment:
We'd like to retain the checks in case the default value is ever removed.  I figured it was ok to reconstruct the vector because this is startup code executed only once and it would be very unusual for the vector size to exceed even 10 elements.

Reply:
my main point was having concise and easy to read code at first. I agree that performance could be neglected here if it adds readability or simplicity.
So if you remove it or at least split that huge `if` statement - that will improve readability.
technically I do not see any difference in functionality if you'll leave it like that:
```c++
if( options.count( "p2p-listen-endpoint" ) ) {
}
```
in that case `p2p_addresses` will be empty anyways and `sort` /`erase` won't crash on empty vector. If I missed something you could just split the `if` like that:
```c++
if( options.count( "p2p-listen-endpoint" ) ) {
   auto p2ps = options.at( "p2p-listen-endpoint" ).as<vector<string>>();
   if (!p2ps.empty() && !p2ps.front().empty()) { 
      p2p_addresses = p2ps;
       /*...*/
   }
}
```

Reply:
Updated for readability.